### DISCUSSION TITLE
Is it possible to allow only authenticated users to "like" an article instead of anonymous authentication?

### DISCUSSION THREAD

**Discussion by mehul-thanki** [nunocoracao/blowfish#1416] [Q&A] [answered]
First of all, Thanks to Nuno! Blowfish is really good and powerful theme. 

I have added Firebase "Views" and "Likes" functions to the site.  

Regarding the "Like" function, I am thinking to protect this action in such a way that a user has to authenticate or at least provide user name or email before liking the post and the system would save the user name/email in the firestore database and remember that user so that next time, the user doesn't have to enter name/email or authenticate. 

Is it a good idea to implement something like this?  Is it possible?  Please advise.

**Comment by nunocoracao** [comment, marked as answer]
Hello @mehul-thanki ,
That is not supported out-of-the-box with Blowfish. You would need to override the firebase integration to use that authentication mechanism and also implement login/signup - which I think Firebase supports a login modal out of the box. 
You can also submit a ticket for me to track this request.

**Reply by mehul-thanki** [reply]
Thanks @nunocoracao!

I'll check out firebase integration and try to implement login/signup. 

I am hosting my site on firebase and with that I tested the view/likes function and overall it works fine, but I noticed below issue. 

After enabling view/likes and deployed on firebase, I accessed the site from 2 unique devices (a mobile and a desktop) and played around with likes. Just after that I checked firebase authentication tab and to my surprise, it created around 375 anonymous users. It should create only 2 anonymous users. I suspect, there is some issue while accessing the site from the mobile where it creates multiple users from the same session.   

I'll create issue from this discussion to track this request.
